# HG changeset patch
# User Colin Macdonald <cbm@m.fsf.org>
# Date 1516041090 28800
#      Mon Jan 15 10:31:30 2018 -0800
# Branch stable
# Node ID 4a4a8b2a5bf2aeca01207d0e0b09bc27da4787d2
# Parent  3e17190dfaea36a024d0cb8cc0911f5b7f3cfda7
Test for incorrect regexprep on ARM platforms (bug #52810).

* regexp.cc (regexprep): Add BIST test.

diff --git a/libinterp/corefcn/regexp.cc b/libinterp/corefcn/regexp.cc
--- a/libinterp/corefcn/regexp.cc
+++ b/libinterp/corefcn/regexp.cc
@@ -1511,9 +1511,13 @@ function.
 
 # Nasty lookbehind expression
 %!test
 %! warning ("off", "Octave:regexp-lookbehind-limit", "local");
 %! assert (regexprep ('x^(-1)+y(-1)+z(-1)=0', '(?<=[a-z]+)\(\-[1-9]*\)', '_minus1'),'x^(-1)+y_minus1+z_minus1=0');
 
 %!assert (regexprep ("\n", '\n', "X"), "X")
 %!assert (regexprep ("\n", "\n", "X"), "X")
+
+## Empty matches were broken on ARM architecture
+%!test <*52810>
+%! assert (strcmp (regexprep ("\nabc", "^(\t*)(abc)$", "$1$2", "lineanchors"), "\nabc"))
 */
